pints/interfaces/_stan.py

```diff
@@ -0,0 +1,101 @@
+#
+# Interface for Stan models
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import numpy as np
+from collections import Counter
+import pystan
+
+from . import InterfaceLogPDF
+
+
+class StanLogPDF(InterfaceLogPDF):
+    def __init__(self, stan_code, stan_data):
+
+        print("Compiling Stan model then test running...")
+        sm = pystan.StanModel(model_code=stan_code)
```

Comment:
It would be good if we can allow for pickling of the compiled model, something like the below (where the user optionally passes in a filename to pickle to). An even better way might be to hash the input code and generate a filename based on that?

```suggestion
        if pickle_filename:
            if os.path.isfile(pickle_filename):
                sm = pickle.load(open(pickle_filename, 'rb'))
            else:
                sm = pystan.StanModel(model_code=stan_code)
                pickle.dump(sm, open(pickle_filename, 'wb'))
        else:
            sm = pystan.StanModel(model_code=stan_code)
```